pkg/controllers/disruption/types.go

```diff
@@ -60,60 +62,31 @@ type Candidate struct {
 }
 
 //nolint:gocyclo
-func NewCandidate(ctx context.Context, kubeClient client.Client, recorder events.Recorder, clk clock.Clock, node *state.StateNode, pdbs *PDBLimits,
+func NewCandidate(ctx context.Context, kubeClient client.Client, recorder events.Recorder, clk clock.Clock, node *state.StateNode, pdbs *pdb.Limits,
 	nodePoolMap map[string]*v1beta1.NodePool, nodePoolToInstanceTypesMap map[string]map[string]*cloudprovider.InstanceType, queue *orchestration.Queue) (*Candidate, error) {
-
-	if node.Node == nil || node.NodeClaim == nil {
-		return nil, fmt.Errorf("state node doesn't contain both a node and a nodeclaim")
-	}
-	// skip any candidates that are already marked for deletion and being handled
-	if node.MarkedForDeletion() {
-		return nil, fmt.Errorf("state node is marked for deletion")
-	}
-	// skip candidates that aren't initialized
-	if !node.Initialized() {
-		return nil, fmt.Errorf("state node isn't initialized")
+	if err := node.IsDisruptable(ctx, kubeClient); err != nil {
```

Comment:
Nice! Love that we are returning this error back now and just using that to fire the event!